### PR TITLE
Add Real Escape Game docs

### PR DESCRIPTION
**Pull request by DaniElectra**
This adds documentation for the custom methods used in DataStore and MatchmakeExtension on Real Esacape Game.

Alongside that, update the UserMessage structure to have the original version and the NEX 4.0 version, and update the
DeliverMessageMultiTarget method on Messaging and Message Delivery with field names from Real Escape Game.

The Xenoblade X page for Message Delivery is removed due to the method being present in more games, and a minor fix on DataStore.

### CODE REVIEW

File: docs/nex/protocols/datastore/index.md [hunk 1]
```diff
@@ -967,7 +968,7 @@ In NEX version 3.0, the data ID was set to a Uint64 and the version was set to a
 | Uint8                  | searchTarget                    |
 | [List]&lt;[PID]&gt;    | ownerIds                        |
 | Uint8                  | ownerType                       |
-| [List]&lt;Uint64&gt;   | destinationIds                  |
+| [List]&lt;[PID]&gt;    | destinationIds                  |
```
Comment (jonbarrow) [inline docs/nex/protocols/datastore/index.md:971]: Is this correct? Super Mario Maker uses this class in several methods, and decoding this as a uint64 has worked fine there? If this field is actually a PID then that would imply that Super Mario Maker uses NEX 4.0+ PIDs, which seems off?

Though I'll have to double check if Super Mario Maker actually populates this field at all. If it's always an empty list then that may be why no issues have cropped up before

File: docs/nex/protocols/datastore/real-escape-game.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+#### Request
+
+| Type   | Name          |
+|--------|---------------|
+| Uint32 | applicationId |
+| Uint32 | key           |
+
+#### Response
+
+This method does not return anything.
```
Comment (jonbarrow) [inline docs/nex/protocols/datastore/real-escape-game.md:65]: Super Mario Maker also has these methods, but the config values are signed ints not unsigned? Is this just a version difference or is one of these docs wrong?

File: docs/nex/protocols/datastore/index.md (hunk 1 above)
Comment (DaniElectra) [reply docs/nex/protocols/datastore/index.md:971]: On all games up to NEX 3.10 (this includes SMM), this field is set as a uint32, but it was updated on NEX 4 to a uint64, so it's very likely this is a PID list that has always been zero on all games we checked.

File: docs/nex/protocols/datastore/real-escape-game.md (hunk 2 above)
Comment (DaniElectra) [reply docs/nex/protocols/datastore/real-escape-game.md:65]: I'm checking both code bins and SMM indeed has the field as a int32, but Real Escape Game uses a uint32.

Comment (jonbarrow) [reply docs/nex/protocols/datastore/real-escape-game.md:65]: Sounds good 👍

File: docs/nex/protocols/datastore/index.md (hunk 1 above)
Comment (jonbarrow) [reply docs/nex/protocols/datastore/index.md:971]: Sounds reasonable 👍

**Pull request merged**